ios/PurchasesHybridCommon/PurchasesHybridCommon/RCCommonFunctionality.m

```diff
@@ -277,22 +266,36 @@ + (void)purchasePackage:(NSString *)packageIdentifier
 
                     if (@available(iOS 12.2, macos 10.14.4, tvOS 12.2, *)) {
                         if (discountTimestamp) {
-                            SKPaymentDiscount *discount = self.discounts[discountTimestamp];
+                            RCStoreProductDiscount *discount = self.discounts[discountTimestamp];
                             if (discount == nil) {
                                 [self productNotFoundErrorWithDescription:@"Couldn't find discount."
                                                             userCancelled:[NSNumber numberWithBool:NO]
                                                                completion:completion];
                                 return;
                             }
-
-                            [RCPurchases.sharedPurchases purchasePackage:aPackage
-                                                            withDiscount:discount
-                                                         completionBlock:completionBlock];
+                            
+                            [RCPurchases.sharedPurchases getPromotionalOfferForProductDiscount:discount
+                                                                                   withProduct:aPackage.storeProduct withCompletion:^(RCPromotionalOffer * _Nullable offer, NSError * _Nullable error) {
+                                
+                                if (offer == nil) {
+                                    [self productNotFoundErrorWithDescription:@"Couldn't find discount."
+                                                                userCancelled:[NSNumber numberWithBool:NO]
```

Comment:
Why is how come this isn't using `Bool`? This should be just 
```suggestion
                                                                userCancelled:NO
```

Reply:
`NO` doesn't work because its a Bool and that methods needs an `NSNumber` but I think I could do `@NO`!